loader/src/Loader.cc

```diff
@@ -536,12 +535,6 @@ namespace ignition
         const std::string& _pathToLibrary) const
     {
       std::vector<Info> loadedPlugins;
-
-      // This function should never be called with a nullptr _dlHandle
-      assert(_dlHandle &&
-             "Bug in code: Loader::Implementation::LoadPlugins was called with "
-             "a nullptr value for _dlHandle.");
```

Comment:
Isn't this safety check needed anymore?

Reply:
We should call dlopen with `nullptr` to get all the symbols from the executable. According with the [documentation](https://linux.die.net/man/3/dlopen):

```txt
If filename is NULL, then the returned handle is for the main program
```

Reply:
But `_dlHandle` is being used in `dlsym` below, not `dlopen` :monocle_face:

Reply:
Sorry, this is the right one:

```
If dlsym() is called with the special handle NULL, it is interpreted as a reference to the executable or shared object from which the call is being made. Thus a shared object can reference its own symbols.
```

Reply:
This sounds like a potential change in behaviour, but I haven't looked closely at this yet.

I noticed you removed the Blueprint / Citadel / Dome labels from this PR and left Edifice. This PR is targeting `ign-plugin1`, so it would apply for all those versions though. Is your intention to target a new major version?

Reply:
There aren't any changes in the API, I think we can target the rest of the versions. This will require to backport the changes in all the "LoadPlugin" libraries for all the versions